runtime/shared_common/OSCache.cpp

```diff
@@ -1285,9 +1295,8 @@ SH_OSCache::generateCacheUniqueID(J9VMThread* currentThread, const char* cacheDi
 	getCacheVersionAndGen(PORTLIB, vm, nameWithVGen, J9SH_MAXPATH, cacheName, &versionData, OSCACHE_CURRENT_CACHE_GEN, true, layer);
 	/* Directory is included here, so if the cache directory is renamed, caches with layer > 0 becomes unusable */
 	getCachePathName(PORTLIB, cacheDirName, cacheFilePathName, J9SH_MAXPATH, nameWithVGen);
-	I_64 timeStamp = j9file_lastmod(cacheFilePathName);
 	I_64 fileSize = j9file_length(cacheFilePathName);
-	return j9str_printf(PORTLIB, buf, bufLen, "%s-%llx_%llx", cacheFilePathName, fileSize, timeStamp);
+	return j9str_printf(PORTLIB, buf, bufLen, "%s-%llx_%llx_%llx_%llx_%llx_%llx", cacheFilePathName, fileSize, createtime, metadataBytes, classesBytes, lineNumTabBytes, varTabBytes);
```

Comment:
This should assert that the data fit into the bufLen and wasn't truncated, as some callers pass in a fixed buffer size.